clang/include/clang/Basic/AttrDocs.td

```diff
@@ -1994,6 +1994,24 @@ can be lowered.
   }];
 }
 
+def SYCLIntelBufferLocationAttrDocs : Documentation {
+  let Category = DocCatFunction;
+  let Heading = "kernel_args_buffer_location";
+  let Content = [{
+The attribute is being implicitly applied to an OpenCL kernel parameters
```

Comment:
This needs some serious word-smithing, but I'll give it a shot.  I'll let the others further clarify.

"This attribute is implicitly added to OpenCL kernel parameters generated from a SYCL kernel object.  It lacks a spelling, as it is not intended to be used by the programmer.

This attribute causes clang to generate metadata on the OpenCL kernel containing the number of kernel parameters.  The metadata contains an integer that is set according to the values passed through the ``accessor`` property ``buffer_location``.  These values are mapped to the actual locations of the buffers (such as DDR, QDR, etc).  The default value is -1.

Reply:
Thanks, I like it! I've also added 'pointer' to 'kernel parameters' in the first string.